dlib/test/dnn.cpp

```diff
@@ -3970,6 +3970,20 @@ namespace
         DLIB_TEST(layer<4>(net).layer_details().get_learning_rate_multiplier() == 0.01);
     }
 
+// ----------------------------------------------------------------------------------------
+
+    template <typename SUBNET>
+    using conblock = relu<bn_con<add_layer<con_<16, 3, 3, 2, 2, 1, 1>, SUBNET>>>;
+
+    void test_input_ouput_mappers()
+    {
+        using net_type = loss_binary_log_per_pixel<con<1, 1, 1, 1, 1,repeat<3, conblock, tag1<input_rgb_image>>>>;
```

Comment:
I don't know why, but I if don't put a tag as the subnet of the `repeat` layer, this fails to compile.

Reply:
Yeah, input layers are special :|

Reply:
I have learned to love them :)